public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+  Therefore, do the following three things in `app.module.ts`:
+  1. Use a JavaScript `import` statement to access 
+  the `ReactiveFormsModule` and the `HeroDetailComponent`.
+  1. Add `ReactiveFormsModule` to the `AppModule`'s `imports` list.
+  1. Add `HeroDetailComponent` to the declarations array.
+
++makeExample('reactive-forms/ts/app/app.module.ts', 'v1','app/app.module.ts (excerpt)')(format=".")
+
+a#update
+.l-main-section
+:marked
+  ## Display the _HeroDetailComponent_
+  Revise the `AppComponent` template so it displays the `HeroDetailComponent`.
++makeExample('reactive-forms/ts/app/app.component.1.ts', '','app/app.component.ts')(format=".")
+
+:marked
+  ### Style the app
+  You used bootstrap CSS classes in the template HTML of both the `AppComponent` and the `HeroDetailComponent`.
+  Add the `bootstrap` _CSS stylesheet_ to the head of `index.html`:
+
++makeExample('reactive-forms/ts/index.html', 'bootstrap','index.html')(format=".")
+
+:marked
+  Now that everything is wired up, the browser should display something like this: 
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/hero-detail.png" width="400px" alt="Basic reactive form")
+
+:marked
+ The form looks great. But does it work? 
+  When the user enters a name, where does the value go? 
+  
+  The value goes into the **_form model_** that backs the group's `FormControls`.
+  To see the form model, add the following line after the 
+  closing `form` tag in the `hero-detail.component.html`:
+
++makeExample('reactive-forms/ts/app/hero-detail-2.component.html', 'form-value-json','app/hero-detail.component.html')(format=".")
+
+:marked
+  The `heroForm.value` returns the _form model_.
+  Piping it through the `JsonPipe` renders the model as JSON in the browser:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/json-output.png" width="400px" alt="JSON output")
+
+:marked
+  The initial `name` property value is the empty string. 
+  Type into the _name_ input box and watch the keystokes appear in the JSON.
+
+
+
+:marked
+  Great! You have the basics of a form. 
+  
+  In real life apps, forms get big fast. 
+  The way you are creating `FormControls` currently is both tedious and hard to maintain. 
+  `FormBuilder` makes form development and maintenance easier.
+
+
+.l-main-section
+a#formbuilder
+:marked
+  ## Introduction to _FormBuilder_
+
+  The `FormBuilder` class helps reduce repetition and 
+  clutter by handling details of control creation for you. 
+
+  You already imported `FormBuilder` into `hero-detail.component.ts`.
+  Use it now to refactor the `HeroDetailComponent` into something that's a little easier to read and write,
+  by following this plan:
+
+  * Explicitly declare the type of the `heroForm` property to be `FormGroup`; you'll initialize it later.
+  * Inject a `FormBuilder` into the constructor.
+  * Add a new `createForm` method that uses the `FormBuilder` to define the `heroForm`.
```

Comment:
How about:

`Add a new ` method that uses the `FormBuilder` to define the `heroForm`. We'll call ours `createMethod`.`